app/middlewares/user.js

```diff
@@ -0,0 +1,36 @@
+const { checkSchema, validationResult } = require('express-validator');
+
+exports.dataUser = (req, res, next) => {
+  const errors = validationResult(req);
+  if (!errors.isEmpty()) {
+    return res.status(422).json({ errors: errors.mapped() });
+  }
+  return next();
+};
+
+exports.schemaUser = checkSchema({
+  password: {
+    exists: true,
+    errorMessage: 'Password is required',
+    isLength: {
+      errorMessage: 'Password should be at least 8 chars long',
+      options: { min: 8 }
+    }
+  },
+  email: {
+    exists: true,
+    errorMessage: 'Email is required',
+    isEmail: {
+      errorMessage: 'Email is invalid',
+      bail: true
+    }
+  },
```

Comment:
Here you can also add the regex that you already use in from the controller.

Reply:
You can also add validation if a user exists, or if it seems more convenient you can add a separate middleware for that.